src/react-components/ui-root.js

```diff
@@ -623,6 +624,28 @@ class UIRoot extends Component {
       this.state.audioTrackClone.stop();
     }
 
+    constraints.audio.echoCancellation = true;
+    constraints.audio.noiseSuppression = true;
+    constraints.audio.autoGainControl = true;
+
+    if (isFirefoxReality) {
+      //workaround for https://bugzilla.mozilla.org/show_bug.cgi?id=1626081
+      constraints.audio.echoCancellation =
+        window.APP.store.state.preferences.disableEchoCancellation === false ? true : false;
+      constraints.audio.noiseSuppression =
+        window.APP.store.state.preferences.disableNoiseSuppression === false ? true : false;
+      constraints.audio.autoGainControl =
+        window.APP.store.state.preferences.disableAutoGainControl === false ? true : false;
```

Comment:
If the user does not set a preference for this, it will be `undefined` (not `false`), which will mean these are all disabled by default. Is this the intended result?

(The `defaultBool` in the preference screen only determines whether the checkbox is checked, but does not actually result in anything being put in the store for that preference. The thinking behind this is to make it so that values in the store can be thought of as being explicitly chosen by the user :
https://github.com/mozilla/hubs/blob/master/src/react-components/preference-list-item.js#L119

We should probably have a simple method for getting the default value for a given preference, so that there's less of a chance of a mismatch between what the list item shows and what the code does.)

Reply:
yes, they are "disable" flags so they should be false by default. Only on FxR will they default to true, which means they are disabled.

Reply:
oh wait i have a typo here

Reply:
I realize this is confusing, the intent is this (for FxR):
e.g. for echo cancellation 
if `disableEchoCancellation` is undefined or TRUE, the constraint should default to false.
if `disableEchoCancellation` is explicitly FALSE, its assumed that the user explicitly wants to enable it anyways, despite being default to off in FxR, so the constraint is set to true.